CMake-related fixes to use correct backend for MKL and allow MKL_CUSTOM=ON

Author: Bart Oldeman <bart.oldeman@calculquebec.ca>
diff -ur flexiblas-3.0.4/src/backends/intel_mkl/mkl_2018.3.list flexiblas-3.0.4.new/src/backends/intel_mkl/mkl_2018.3.list
--- flexiblas-3.0.4/src/backends/intel_mkl/mkl_2018.3.list	2020-10-22 09:07:49.000000000 +0000
+++ flexiblas-3.0.4.new/src/backends/intel_mkl/mkl_2018.3.list	2021-09-28 17:35:28.554563411 +0000
@@ -56,6 +56,7 @@
 dasum_
 daxpy_
 daxpyi_
+dcabs1_
 dcopy_
 ddot_
 ddoti_
@@ -106,6 +107,7 @@
 sasum_
 saxpy_
 saxpyi_
+scabs1_
 scasum_
 scnrm2_
 scopy_
diff -ur flexiblas-3.0.4/src/CMakeLists.txt flexiblas-3.0.4.new/src/CMakeLists.txt
--- flexiblas-3.0.4/src/CMakeLists.txt	2020-10-22 09:07:49.000000000 +0000
+++ flexiblas-3.0.4.new/src/CMakeLists.txt	2021-09-28 17:28:59.862127651 +0000
@@ -331,7 +331,7 @@
 INCLUDE(CheckFunctionExists)
 
 # SET(BLA_STATIC TRUE)
-MACRO(BuildBackendBlas VENDOR BACKEND_CODE)
+MACRO(BuildBackendBlas VENDOR BACKEND_CODE_PARAM)
     MESSAGE(STATUS "Build additional BLAS interface: ${VENDOR}")
     MESSAGE(STATUS "-- Libs: ${${VENDOR}_LIBRARY}")
     MESSAGE(STATUS "-- Linker Flags: ${${VENDOR}_LINKER_FLAGS}")
@@ -356,6 +356,8 @@
     # Use the MKL Backend Code if it is MKL
     IF(_MKL_VERSION)
         SET(BACKEND_CODE backends/intel_mkl/backend.c)
+    ELSE()
+        SET(BACKEND_CODE ${BACKEND_CODE_PARAM})
     ENDIF()
 
     SET(CMAKE_REQUIRED_LIBRARIES ${__CMAKE_REQUIRED_LIBRARIES})
@@ -672,6 +674,8 @@
 	TARGET_LINK_LIBRARIES(flexiblas_intel_mkl_seq ${MKL_SEQUENTIAL})
 	ADD_DEPENDENCIES(flexiblas_intel_mkl mkl_parallel)
 	ADD_DEPENDENCIES(flexiblas_intel_mkl_seq mkl_sequential)
+	SET(INTEL_MKL_BLAS_LIBRARIES flexiblas_intel_mkl PARENT_SCOPE)
+	SET(INTEL_MKL_SEQ_BLAS_LIBRARIES flexiblas_intel_mkl_seq PARENT_SCOPE)
 
 	LIST(APPEND EXTRA_BLAS "INTEL_MKL")
 	LIST(APPEND EXTRA_BLAS "INTEL_MKL_SEQ")
